src/frog.js

```diff
@@ -0,0 +1,360 @@
+const BtpPacket = require('btp-packet')
+const IlpPacket = require('ilp-packet')
+const crypto = require('crypto')
+
+function generateRequestId () {
+  const buf = crypto.randomBytes(4)
+  const hex = buf.toString('hex')
+  return parseInt(hex, 16)
+}
+
+function MakeProtocolData (obj) {
+  let protocolData = []
+  if (obj.from) {
+    protocolData.push({
+      protocolName: 'from',
+      contentType: BtpPacket.MIME_TEXT_PLAIN_UTF8,
+      data: Buffer.from(obj.from, 'ascii')
```

Comment:
It should probably check if these fields are present, no?

Reply:
wait, what do you mean, that's checked in line 13, right?